tasks/update.py

```diff
@@ -0,0 +1,292 @@
+        self.logger.info("Following line items will not be updated:")
+        table.field_names = [
+            f"{self.color.BOLD}Line Item Name{self.color.END}",
+            f"{self.color.BOLD}Reason{self.color.END}",
+        ]
+        for line_item,reason in skip_line_items.items():
+            table.add_row([
+                f"{self.color.BLUE}{line_item}{self.color.END}",
+                f"{self.color.BLUE}{reason}{self.color.END}",
+            ])
+        self.logger.info(table)
+
+    def print_line_items_with_current_position(self, line_items_to_update):
+        """
+        Function to print the line items that will be updated by script along with current-video-position and new-video-position
+        """
+        if len(line_items_to_update) <= 0:
+            return
+
+        table = PrettyTable()
+        self.logger.info("Following line items will be updated:")
+        table.field_names = [
+            f"{self.color.BOLD}Line Item Name{self.color.END}",
+            f"{self.color.BOLD}Current Video Position{self.color.END}",
+            f"{self.color.BOLD}New Video Position{self.color.END}"
+        ]
+        for line_item,current_position in line_items_to_update.items():
+            table.add_row([
+                f"{self.color.BLUE}{line_item}{self.color.END}",
+                f"{self.color.BLUE}{current_position}{self.color.END}",
+                f"{self.color.BLUE}{self.setting_class.NEW_VIDEO_POSITION}{self.color.END}"
+            ])
+        self.logger.info(table)
+
+
+
+    def select_line_items_to_update(self,line_items):
+        """
+        Selects line items to update based on specified criteria.
+        Parameters:
+            line_items (list): List of line items to be evaluated for updates.
+        Returns:
+                1. List of line items to be updated.
+                2. Dictionary mapping line item names to their current targeted video positions.
+                3. Dictionary containing line item names and reasons for skipping (not updating) each line item.
+        """
+        # List of line items to be updated
+        updated_line_items = []
+        # Dictionary mapping line item names to their current targeted video positions.
+        line_items_with_current_position = {}
+        # Dictionary containing line item names and reasons for skipping (not updating) each line item.
+        skip_line_items = {}
+
+        for line_item in line_items:
+            # do not update line-item if targeting information is missing
+            if 'targeting' not in line_item or not line_item['targeting']:
+                skip_line_items[line_item['name']] = "targeting information is missing"
+                continue
+
+            # do not update line-item if targeting.requestPlatformTargeting (Inventory type) information is missing
+            if 'requestPlatformTargeting' not in line_item['targeting'] or not line_item['targeting']['requestPlatformTargeting']:
```

Comment:
As `videoPositionTargeting` comes directly under `targeting` object, do we need do reject LI based on this condition?

Ref: https://developers.google.com/ad-manager/api/reference/v202311/AdRuleService.VideoPositionTargeting

Reply:
as discussed, this update utility should be used only for updating line items created by our tool.
In our tool, while creating line items we always set the value for targeting.requestPlatformTargeting hence we will not add this conditions in code